File: week6/readability/readability.py
import cs50
from cs50 import get_string


def main():
    # Prompt for text input
    txt_input = get_string("Text: ")
    # Note the length of the text
    txt_len = len(txt_input)
    
    # Initialize l, w, s and i
    letter = 0
    # Split the text into words and count with len()
    word = len(txt_input.split())
    sent = 0
    i = 0
    
    # Loop itiration until done (length of text input)
    while i < txt_len:
        # If alphabetic add letters
        if txt_input[i].isalpha():
            letter += 1
        # If end of sentence add sent
        if txt_input[i] == "." or txt_input[i] == "!" or txt_input[i] == "?":
            sent += 1
        i += 1
        
    # Apply grading formula and round up 
    L = 100 * letter / word
    S = 100 * sent / word
    grade = round((0.0588 * L) - (0.296 * S) - 15.8)

    # Check grade and print out
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


if __name__ == "__main__":
    main()                

File: week2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>


int main(void)
{
    //Prompt userinput for a text
    string text = get_string("Text: ");
    int count_letter = 0;
    int count_word = 1;
    int count_sent = 0;


    for (int i = 0; i < strlen(text); i++)
    {
        //Count letters
        if (isalpha(text[i]))
        {
            count_letter += 1;
        }
        //Count words
        else if (isspace(text[i]) && isalpha(text[i + 1]))
        {
            count_word += 1;
        }
        // Count sentences
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?' || text[i] == ':')
        {
            count_sent += 1;
        }
    }

    //Count averages
    float L = 100 * (float) count_letter / (float) count_word;
    float S = 100 * (float) count_sent / (float) count_word;

    //Apply formula
    float index = (0.0588 * L) - (0.296 * S) - 15.8;
    float grade = round(index);

    //Print grade results
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (1 <= grade && grade <= 16)
    {
        printf("Grade %i\n", (int) grade);
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }

}




